yb-voyager/cmd/endMigrationCommand.go

```diff
@@ -493,13 +493,65 @@ func cleanupTargetDB(msr *metadb.MigrationStatusRecord) {
 		utils.ErrExit("clearing migration state from target db: %v", err)
 	}
 
+	fmt.Println("deleting yb replication slot and publication")
+	sourceYB := srcdb.Source{
+		DBType:         tconf.TargetDBType,
+		Host:           tconf.Host,
+		Port:           tconf.Port,
+		User:           tconf.User,
+		Password:       tconf.Password,
+		DBName:         tconf.DBName,
+		Schema:         tconf.Schema,
+		SSLMode:        tconf.SSLMode,
+		SSLCertPath:    tconf.SSLCertPath,
+		SSLKey:         tconf.SSLKey,
+		SSLRootCert:    tconf.SSLRootCert,
+		SSLCRL:         tconf.SSLCRL,
+		SSLQueryString: tconf.SSLQueryString,
+		Uri:            tconf.Uri,
+	}
+	err = sourceYB.DB().Connect()
+	if err != nil {
+		utils.ErrExit("connecting to YB as source db for deleting replication slot and publication: %v", err)
+	}
+	defer sourceYB.DB().Disconnect()
+	err = deleteYBReplicationSlotAndPublication(msr, sourceYB)
+	if err != nil {
+		utils.ErrExit("deleting yb replication slot and publication: %v", err)
+	}
+
 	if msr.YBCDCStreamID == "" {
 		log.Info("yugabytedb cdc stream id is not set. skipping deleting stream id")
 		return
 	}
 	deleteCDCStreamIDForEndMigration(tconf)
 }
 
+func deleteYBReplicationSlotAndPublication(msr *metadb.MigrationStatusRecord, source srcdb.Source) (err error) {
```

Comment:
let's make the signature of this function be (replicationSlotName, publicationName, source)